src/libs/blueprint/conduit_blueprint_mesh_utils.cpp

```diff
@@ -342,6 +342,326 @@ find_domain_id(const Node &node)
     return domain_id;
 }
 
+
+//-----------------------------------------------------------------------------
+NDIndex::NDIndex(const Node& idx) :
+    NDIndex(&idx)
```

Comment:
Q about design of NDIndex -- What cases do we need to allocate nodes for?

Can re-rework this to rely on fully external data?

We want these to be as light weight as possible.